File: src/main/java/net/hectus/neobb/util/Colors.java
package net.hectus.neobb.util;

import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;

import static net.kyori.adventure.text.format.TextColor.color;

/**
 * Colors to be used instead of {@link NamedTextColor} to ensure a uniform style.
 * Some of these use colors from the original game and some are completely custom.
 */
public final class Colors {
    /** Primary accent color. */ public static final TextColor ACCENT = color(0xB47ACE);
    /** Secondary accent color. */ public static final TextColor SECONDARY = color(0xCE2440);
    /** Pastel Green. */ public static final TextColor GREEN = color(0x5CE673);
    /** Pastel Yellow. */ public static final TextColor YELLOW = color(0xDAE65C);
    /** Pastel Red. */ public static final TextColor RED = color(0xE65C67);
    /** Pastel Blue. */ public static final TextColor BLUE = color(0x6CAAD9);

    // Got these from screenshots of the original game.
    /** {@link NamedTextColor#GRAY Vanilla Gray} (Person Mode). */ public static final TextColor PERSON_0 = NamedTextColor.GRAY;
    /** Dark Green (Person Mode). */ public static final TextColor PERSON_1 = color(0x328825);
    /** Green (Person Mode). */ public static final TextColor PERSON_2 = color(0x37BF1F);
    /** Light Green (Person Mode). */ public static final TextColor PERSON_3 = color(0x6AF95D);
    /** Desaturated Cyan (Person Mode). */ public static final TextColor PERSON_4 = color(0x44D0B9);

    /** Pastel Green, see {@link #GREEN}. */ public static final TextColor POSITIVE = GREEN;
    /** Pastel Yellow, see {@link #YELLOW}. */ public static final TextColor NEUTRAL = YELLOW;
    /** Pastel Red, see {@link #RED}. */ public static final TextColor NEGATIVE = RED;
    /** Pastel Blue, see {@link #BLUE}. */ public static final TextColor LINK = BLUE;

    /** {@link NamedTextColor#DARK_GRAY Vanilla Dark Gray}. */ public static final TextColor EXTRA = NamedTextColor.DARK_GRAY;
    /** {@link NamedTextColor#WHITE Vanilla White}. */ public static final TextColor RESET = NamedTextColor.WHITE;
}

File: src/main/kotlin/net/hectus/neobb/util/Colors.kt
package net.hectus.neobb.util

import net.hectus.neobb.util.Colors.BLUE
import net.hectus.neobb.util.Colors.GREEN
import net.hectus.neobb.util.Colors.RED
import net.hectus.neobb.util.Colors.YELLOW
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor

/**
 * Colors to be used instead of [NamedTextColor] to ensure a uniform style.
 * Some of these use colors from the original game and some are completely custom.
 */
object Colors {
    /** Primary Accent-Color (#B47ACE) */
    val ACCENT: TextColor = TextColor.color(0xB47ACE)

    /** Secondary Accent-Color (#CE2440) */
    val SECONDARY: TextColor = TextColor.color(0xCE2440)


    /** Pastel Green (#5CE673) */
    val GREEN: TextColor = TextColor.color(0x5CE673)

    /** Pastel Green (#DAE65C) */
    val YELLOW: TextColor = TextColor.color(0xDAE65C)

    /** Pastel Red (#E65C67) */
    val RED: TextColor = TextColor.color(0xE65C67)

    /** Pastel Blue (#6CAAD9) */
    val BLUE: TextColor = TextColor.color(0x6CAAD9)


    /** [Vanilla Gray][NamedTextColor]. Picked from screenshots of the original game. */
    val PERSON_0: TextColor = NamedTextColor.GRAY

    /** Dark and desaturated green (#328825). Picked from screenshots of the original game. */
    val PERSON_1: TextColor = TextColor.color(0x328825)

    /** Slightly darkened green (#37BF1F). Picked from screenshots of the original game. */
    val PERSON_2: TextColor = TextColor.color(0x37BF1F)

    /** Light and slightly pastel green (#6AF95D). Picked from screenshots of the original game. */
    val PERSON_3: TextColor = TextColor.color(0x6AF95D)

    /** Slightly desaturated cyan (#44D0B9). Picked from screenshots of the original game. */
    val PERSON_4: TextColor = TextColor.color(0x44D0B9)


    /** Positive color to be used for confirmation messages. See [GREEN]. */
    val POSITIVE: TextColor = GREEN

    /** Neutral color to be used for all types of messages. See [YELLOW]. */
    val NEUTRAL: TextColor = YELLOW

    /** Negative color to be used for failure messages. See [RED]. */
    val NEGATIVE: TextColor = RED

    /** Color for links. See [BLUE]. */
    val LINK: TextColor = BLUE


    /** Dark gray color for unimportant text. See [NamedTextColor.DARK_GRAY]. */
    val EXTRA: TextColor = NamedTextColor.DARK_GRAY

    /** Utility color for resetting text back to white. See [NamedTextColor.WHITE]. */
    val RESET: TextColor = NamedTextColor.WHITE
}
